packages/objectloader2/src/operations/downloader.spec.ts

```diff
@@ -0,0 +1,83 @@
+import { beforeEach, describe, expect, test } from 'vitest'
+import createFetchMock from 'vitest-fetch-mock'
+import { vi } from 'vitest'
+import AsyncGeneratorQueue from '../helpers/asyncGeneratorQueue.js'
+import { Item } from '../types/types.js'
+import Downloader from './downloader.js'
+import CacheDatabase from './database.js'
+
+const fetchMocker = createFetchMock(vi)
+
+// sets globalThis.fetch and globalThis.fetchMock to our mocked version
+fetchMocker.enableMocks()
```

Comment:
why not just make fetch injectable through the constructor?

Reply:
no idea....I don't know what's best practice anymore.  This patching was what was hot back in the day. 

I was going to ask you to a meeting to show you what I did to test things.  Sounds like you're suggesting a more typed/DI approach of wrapping fetch in a more basic type and mocking it to test ability

Reply:
you can still mock it, I don't think you necessarily need to wrap it in anything, just inject it in through DI instead of mocking the global fetch import